src/main/java/com/tananaev/jsonpatch/operation/AbsOperation.java

```diff
@@ -18,10 +17,24 @@ public AbsOperation(){
 
     public abstract String getOperationName();
 
-    public abstract JsonElement apply( JsonElement original );
-
-    protected JsonElement duplicate(JsonElement original) {
-        return new JsonParser().parse(original.toString());
-    }
+    /**
+     * Applies the operation on the source and returns the patched object. The source is unmodified.
+     * @param sourceElement to which the patch is to be applied
+     * @return final json after patch is applied
+     */
+    public JsonElement apply( JsonElement sourceElement ){
```

Comment:
Exposed two functions.
_apply_ - Doesn't change input and copies the actual element.
_applyInPlace_ - Modifies input without copying
The _apply_ functionality is the same as it is now and doesn't modify the input. So, any existing callers of _apply_ should not see any changes.